mlir/include/mlir/Dialect/EmitC/IR/EmitC.td

```diff
@@ -288,6 +288,7 @@ def EmitC_CastOp : EmitC_Op<"cast",
   let arguments = (ins EmitCType:$source);
   let results = (outs EmitCType:$dest);
   let assemblyFormat = "$source attr-dict `:` type($source) `to` type($dest)";
```

Comment:
Please add the reference attribute as `arguments` (UnitAttr) into the ODS.
Then we can do a fancy assembly format like
```
emitc.cast %arg0  : i32 to i32&
```
or
```
emitc.cast %arg0 : i32 to i32 as reference
```

Reply:
I'd like to have a single, unified attribute that would apply to both function arguments and ops like these for which we can tag the result as a reference. An interface doesn't look like a good answer as it wouldn't apply to function arguments. Maybe there's something out there already for this?